libr/core/cmd_search.c

```diff
@@ -908,10 +908,69 @@ static RList* construct_rop_gadget(RCore *core, ut64 addr, ut8 *buf, int idx,
 	return hitlist;
 }
 
+static int rop_classify_nops (RCore *core, RList *ropList) {
+	char *cmd, *esil_str;
+	int changes = 1;
+	RListIter *iter, *iter_r;
+	RRegItem *item;
+	RList *head;
+	RHashTable *ht_old = r_hashtable_new ();
+	RHashTable *ht_new = r_hashtable_new ();
+
+	srand (time (NULL));
+
+	// TODO: this seems useless ? should confirm
+	// RReg *hack = core->dbg->reg;
+	// core->dbg->reg = core->anal->reg;
+
+	r_list_foreach (ropList, iter_r, esil_str) {
+
+		// Don't take 'ret' into account
+		if (strcmp (esil_str, "esp,[4],eip,=,4,esp,+=") == 0) {
```

Comment:
asuming this expressiona a RET is a very bad idea. to start with it will only work on x86-32. and this code must be completely arch-independent. you should just check the op->type here